core/policy-monitor/policy-monitor-core/src/main/java/org/eclipse/edc/connector/policy/monitor/manager/PolicyMonitorManagerImpl.java

```diff
@@ -0,0 +1,136 @@
+import org.eclipse.edc.connector.policy.monitor.spi.PolicyMonitorEntry;
+import org.eclipse.edc.connector.policy.monitor.spi.PolicyMonitorEntryStates;
+import org.eclipse.edc.connector.policy.monitor.spi.PolicyMonitorManager;
+import org.eclipse.edc.connector.policy.monitor.spi.PolicyMonitorStore;
+import org.eclipse.edc.connector.spi.contractagreement.ContractAgreementService;
+import org.eclipse.edc.connector.spi.transferprocess.TransferProcessService;
+import org.eclipse.edc.policy.engine.spi.PolicyContextImpl;
+import org.eclipse.edc.policy.engine.spi.PolicyEngine;
+import org.eclipse.edc.spi.query.Criterion;
+import org.eclipse.edc.statemachine.Processor;
+import org.eclipse.edc.statemachine.ProcessorImpl;
+import org.eclipse.edc.statemachine.StateMachineManager;
+
+import java.time.Instant;
+import java.util.function.Function;
+
+import static org.eclipse.edc.connector.policy.monitor.spi.PolicyMonitorEntryStates.STARTED;
+import static org.eclipse.edc.spi.persistence.StateEntityStore.hasState;
+
+public class PolicyMonitorManagerImpl extends AbstractStateEntityManager<PolicyMonitorEntry, PolicyMonitorStore>
+        implements PolicyMonitorManager {
+
+    private PolicyEngine policyEngine;
+    private TransferProcessService transferProcessService;
+    private ContractAgreementService contractAgreementService;
+
+    private PolicyMonitorManagerImpl() {
+
+    }
+
+    @Override
+    protected StateMachineManager.Builder configureStateMachineManager(StateMachineManager.Builder builder) {
+        return builder
+                .processor(processEntriesInState(STARTED, this::processMonitoring));
+    }
+
+    @Override
+    public void startMonitoring(String transferProcessId, String contractId) {
+        var entry = PolicyMonitorEntry.Builder.newInstance()
+                .id(transferProcessId)
+                .contractId(contractId)
+                .traceContext(telemetry.getCurrentTraceContext())
+                .build();
+
+        entry.transitionToStarted();
+
+        update(entry);
+    }
+
+    private boolean processMonitoring(PolicyMonitorEntry entry) {
+        var contractAgreement = contractAgreementService.findById(entry.getContractId());
+        if (contractAgreement == null) {
+            entry.transitionToFailed("ContractAgreement %s does not exist".formatted(entry.getContractId()));
+            update(entry);
+            return true;
+        }
+
+        var policy = contractAgreement.getPolicy();
+        var policyContext = PolicyContextImpl.Builder.newInstance()
+                .additional(Instant.class, Instant.now(clock))
+                .additional(ContractAgreement.class, contractAgreement)
+                .build();
+
+        var result = policyEngine.evaluate("transfer.process", policy, policyContext);
+        if (result.failed()) {
+            monitor.debug(() -> "[policy-monitor] Policy evaluation for TP %s failed: %s".formatted(entry.getId(), result.getFailureDetail()));
```

Comment:
I would use the non-lambda version of .debug(). Also, we may want to (in another PR) look at providing a typed builder for these types of messages where prefixes (e.g. [policy-monitor]) are segmented across implementations

Reply:
Calling the lambda version will avoid to evaluate the string format, that is something we want to avoid whenever the debug logs are disabled (the lambda method is called anyway [by default by the super method](https://github.com/eclipse-edc/Connector/blob/f981237b50d63fa07e7a1a4edd1b7fa2e69e1d79/spi/common/core-spi/src/main/java/org/eclipse/edc/spi/monitor/Monitor.java#L57-L59)). 

I pretty much agree for the "typed builder", I will provide an issue for it.)

Reply:
Yep sorry, didn't see the format